### PR TITLE
adds Redux to the project and combineReducers to index.js of reducers folder

### PR DESCRIPTION
**Pull request by rickmansfield**
Co-Authored-by: Nicholas Hamilton <nhamilton1@users.noreply.github.com>
Co-authored-by: Richard J Oh <ayv8er@users.noreply.github.com>
Co-authored-by: Christopher Boyle <Chris-Boyle-94@users.noreply.github.com>
Co-authored-by: Rafael Mendoza <rafaelmendozajr94@gmail.com>
Co-authored-by: Kade Griffith <cyberkade@users.noreply.github.com>

## Description

Adds redux to the project. Specifically wraps the project base index.js with <Provider>, store, and rootReducer enabling scalability for the future. 

Fixes # Advanced State Management options

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

https://www.loom.com/share/d934833fbf134d2fbbe7748bebcd6bb5

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (nhamilton1) [inline .vscode/settings.json:1]: We should probably delete this file before adding it to the main.

File: src/state/reducers/index.js [hunk 2]
```diff
@@ -1,4 +1,11 @@
-// import all of your reducers into this file, and export them back out. 
+// import all of your reducers into this file, and export them back out.
 // This allows for the simplification of flow when importing reducers into your actions throughout your app.
+// import { combineReducers } from 'redux';
```
Comment (nhamilton1) [inline src/state/reducers/index.js:3]: Can delete this line of commented code.

**Review by nhamilton1 — COMMENTED** [review]
Great boilerplate setup for redux, excited to work more with this!

**Review by Retrofitt — APPROVED** [review]
Setting up redux now really sets this application up for success moving forward and in my opinion is a great addition to the overall architecture of the application.

File: src/state/reducers/index.js (hunk 2 above)
Comment (rickmansfield) [reply src/state/reducers/index.js:3]: I'm actually utilizing that code in my next PR

**Review by teresafranxman97 — APPROVED** [review]
I think it's very important to be able to use state only where it's needed and Redux certainly helps to avoid unnecessary prop drilling. Great thinking!

File: src/state/reducers/index.js (hunk 2 above)
Comment (nhamilton1) [reply src/state/reducers/index.js]: Ah, got it!

File: src/index.js [hunk 3]
```diff
@@ -24,10 +24,19 @@ import { Schedule } from './components/pages/Schedule/Schedule';
 import { ManageResources } from './components/pages/ManageResources/ManageResources';
 import { MenteesProgress } from './components/pages/MenteesProgress/MenteesProgress';
 
+import { createStore, applyMiddleware } from 'redux';
+import { Provider } from 'react-redux';
+import rootReducer from './state/reducers';
+import promiseMiddleware from 'redux-promise';
```
Comment (paulstgermain) [inline src/index.js:30]: Interesting - I'm used to seeing `redux-thunk` brought in at this stage, and actually had to go do some reading to learn more about `redux-promise`! Well done, and thanks for keeping me on my toes! 😁

**Pull request merged**